torch/_functorch/aot_autograd.py

```diff
@@ -172,18 +174,22 @@ def prehook(grad_output):
                 callback_set = True
 
             fx_traceback.set_stack_trace(stack_)
+            if stack_ != prev_stack and max_seq_id >= 0:
+                fx_traceback.set_bwd_seq_id(max_seq_id)
```

Comment:
is it guaranteed that the backward ops will get called in exactly the reverse order compared to the forward ops? I'm not sure this is true...

I think it's also possible to find ops that will be decomposed in the dispatcher ("CompositeImplicitAutograd" ops), so that bwd_seq_id might become < -1 in some cases.

Reply:
That's a good question, I'm not certain I can say guaranteed.  I think the idea is that the bwd pass is just walking the fwd graph in reverse order.  I think this is the assumption used in autograd but I'm not sure how to prove that it is always the case.